src/actions/actions.ts

```diff
@@ -1873,6 +1874,22 @@ class CommandCenterScroll extends BaseCommand {
 }
 
 @RegisterAction
+class CommandCenterScrollMultipleLines extends BaseCommand {
+  modes = [ModeName.VisualLine, ModeName.VisualBlock];
+  keys = ["z", "z"];
+
+  public async exec(position: Position, vimState: VimState): Promise<VimState> {
+    // In these modes, you want to take the full selection and find the middle point
```

Comment:
Apparently I made up functionality, I am not sure how I tested it but it was wrong. Thanks for carefully checking what I am actually doing....